### PR TITLE
Azure Translator 1.0.0

### PR DESCRIPTION
**Pull request by Muennighoff**
TODO:
- [x] Finalize Parallelizer

### CODE REVIEW

File: python-lib/azure_translation_api_client.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        api_key: Azure Translator API Key
+        location: Azure location, e.g. francecentral
+
+    """
+
+    def __init__(self, api_key, location) -> None:
+        self.api_key = api_key
+        self.location = location
+
+        self.endpoint = "https://api.cognitive.microsofttranslator.com/translate"
```
Comment (mhham) [inline python-lib/azure_translation_api_client.py]: Does this change depending on the client or is it always the same ?
If it is a constant add it the the section above below API_EXCEPTIONS

File: python-lib/azure_translation_api_client.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            HTTPError: An error occured accessing the API.
+        """
+        response = requests.post(
+            url=self.endpoint,
+            headers={
+                "Ocp-Apim-Subscription-Key": self.api_key,
+                "Ocp-Apim-Subscription-Region": self.location,
+                "Content-type": "application/json",
+            },
+            params={"api-version": "3.0", "from": source_language, "to": target_language},
```
Comment (mhham) [inline python-lib/azure_translation_api_client.py]: Maybe we should keep the api-version as a constant somewhere to keep track of this. In the constant definitions section ?

**Review by mhham — APPROVED** [review]
LGTM except minor comments

File: python-lib/azure_translation_api_client.py (hunk 1 above)
Comment (Muennighoff) [reply python-lib/azure_translation_api_client.py]: I assume it is always the same as it is used in their official tutorial -- https://docs.microsoft.com/en-us/azure/cognitive-services/Translator/quickstart-translator?pivots=programming-language-javascript&tabs=python#headers

Added it as a constant

File: python-lib/azure_translation_api_client.py (hunk 2 above)
Comment (Muennighoff) [reply python-lib/azure_translation_api_client.py]: Good idea done

File: python-lib/azure_translation_api_client.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import requests
+
+# ==============================================================================
+# CONSTANT DEFINITION
+# ==============================================================================
+
+
+API_EXCEPTIONS = (requests.HTTPError,)
+AZURE_ENDPOINT = "https://api.cognitive.microsofttranslator.com/translate"
+AZURE_VERSION = "3.0"
```
Comment (mhham) [inline python-lib/azure_translation_api_client.py]:
```suggestion
AZURE_TRANSLATION_API_VERSION = "3.0"
```

**Review by mhham — APPROVED** [review]

**Pull request merged**